### PR TITLE
api: custom jsonschema valitation

### PR DESCRIPTION
**Pull request by egabancho**
* Allows usage of custom jsonschema validation class. (addresses ##146)

cc @jirikuncar @lnielsen

### CODE REVIEW

File: invenio_records/api.py [hunk 1]
```diff
@@ -76,33 +76,65 @@ def updated(self):
...
-
-        returns ``None``, which means that the validation was successful, while
-
-        >>> validate('bar', {'format': 'foo'}, format_checker=checker)
-        Traceback (most recent call last):
-        ...
-        ValidationError: 'bar' is not a 'foo'
-        ...
+        :Keyword Arguments:
+          * **format_check** --
```
Comment (jirikuncar) [inline invenio_records/api.py]: format_checker

File: invenio_records/api.py [hunk 2]
```diff
@@ -76,33 +76,65 @@ def updated(self):
...
+
+            >>> from jsonschema import FormatChecker
+            >>> from jsonschema.validators import validate
+            >>> checker = FormatChecker()
+            >>> checker.checks('foo')(lambda el: el.startswith('foo'))
+            <function <lambda> at ...>
+            >>> validate('foo', {'format': 'foo'}, format_checker=checker)
+
+            returns ``None``, which means that the validation was successful,
+            while
```
Comment (jirikuncar) [inline invenio_records/api.py]: while what? It creates an expectation that the sentence finishes after the code snippet.

File: invenio_records/api.py [hunk 3]
```diff
@@ -76,33 +76,65 @@ def updated(self):
...
+            A :class:`jsonschema.IValidator` class that will be used to
+            validate. It will be used as `cls` argument when calling
+            :func:`jsonschema.validate`. For example
+
+            >>> from jsonschema.validators import extend, Draft4Validator
+            >>> NoRequiredValidator = extend(
+                    Draft4Validator,
+                    validators={'required': lambda v, r, i, s: None})
+            >>> schema = {
+                    'type': 'object',
```
Comment (jirikuncar) [inline invenio_records/api.py]: I used to have a problem in the past when I didn't start multiline with `...`

File: invenio_records/api.py [hunk 4]
```diff
@@ -76,33 +76,65 @@ def updated(self):
...
+            while
+
+            >>> validate('bar', {'format': 'foo'}, format_checker=checker)
+            Traceback (most recent call last):
+            ...
+            ValidationError: 'bar' is not a 'foo'
+            ...
+
+          * **validator** --
+            A :class:`jsonschema.IValidator` class that will be used to
```
Comment (jirikuncar) [inline invenio_records/api.py]: used to validate -> used for validating the record, used for record validation, ... ?

File: invenio_records/api.py (hunk 2 above)
Comment (egabancho) [inline invenio_records/api.py]: It was the mid season cliffhanger 😉

**Comment by jirikuncar** [comment]
@egabancho `s/(addresses ##146)/(addresses #146)/`

**Pull request merged**